os/drivers/lcd/st7785_st7701.c

```diff
@@ -0,0 +1,420 @@
+#ifndef CONFIG_LCD_NOGETRUN
+	pinfo->getrun = &lcd_getrun;	/* Get a run from LCD memory */
+#endif
+	return OK;
+}
+
+/****************************************************************************
+ * Name:  lcd_getpower
+ ****************************************************************************/
+
+static int lcd_getpower(FAR struct lcd_dev_s *dev, int color)
+{
+	return OK;
+}
+
+/****************************************************************************
+ * Name:  lcd_setpower
+ ****************************************************************************/
+
+static int lcd_setpower(FAR struct lcd_dev_s *dev, int power)
+{
+	FAR struct st7785_st7701_dev_s *priv = (FAR struct st7785_st7701_dev_s *)dev;
+	if (power > CONFIG_LCD_MAXPOWER) {
+		printf("Power exceeds CONFIG_LCD_MAXPOWER %d", CONFIG_LCD_MAXPOWER);
+		return -1;
+	}
+	if (power > 0) {
+                priv->config->backlight(power);
+                priv->power = power;
+        } else {
+		priv->config->backlight(power);
+                priv->power = 0;
+        }
+	return OK;
+}
+
+/****************************************************************************
+ * Name:  lcd_init
+ *
+ * Description:
+ *   init the lcd
+ *
+ ****************************************************************************/
+
+static int lcd_init(FAR struct lcd_dev_s *dev)
+{
+	FAR struct st7785_st7701_dev_s *priv = (FAR struct st7785_st7701_dev_s *)dev;
+	priv->config->reset();
+	if (send_init_cmd(priv->dsi_dev, lcd_init_cmd_g) == OK) {
+                printf("LCD Init sequence completed\n");
+        } else {
+                printf("ERROR: LCD Init sequence failed\n");
+        }
+	priv->config->init();
+	priv->config->lcd_enable();
+	return OK;
+}
+
+/****************************************************************************
+ * Name:  lcd_setorientation
+ *
+ * Description:
+ *   Set LCD orientation(PORTRAIT/LANDSCAPE).
+ *
+ ****************************************************************************/
+
+static int lcd_setorientation(FAR struct lcd_dev_s *dev, unsigned int orientation)
+{
+	return OK;
+}
+
+/****************************************************************************
+ * Name:  lcd_getcontrast
+ *
+ * Description:
+ *   Get the current contrast setting (0-CONFIG_LCD_MAXCONTRAST).
+ *
+ ****************************************************************************/
+
+static int lcd_getcontrast(FAR struct lcd_dev_s *dev)
+{
+	return OK;
+}
+
+/****************************************************************************
+ * Name:  lcd_setcontrast
+ *
+ * Description:
+ *   Set LCD panel contrast (0-CONFIG_LCD_MAXCONTRAST).
+ *
+ ****************************************************************************/
+
+static int lcd_setcontrast(FAR struct lcd_dev_s *dev, unsigned int contrast)
+{
+	return OK;
+}
+
+FAR struct lcd_dev_s *st7785_st7701_lcdinitialize(FAR struct mipi_dsi_device *dsi, struct st7785_st7701_config_s *config)
```

Comment:
hmm.. when we add new mipi LCD support, will you add it here again like st7785_st7701_stXXX_lcdinitialize? If not, how about using some common name for this type?
Except init command, I can't find any lcd specific code. I think we can change this mipi_lcd.c(?) and separate st7701.c and st7785.c file and then include the mipi_lcd.c file from them. (or just adding mipi_lcd.c file in Make.defs with those lcds.